### PR TITLE
AddVolume and RemoveVolume subresource endpoints

### PR DESCRIPTION
**Pull request by davidvossel**
These are subresource endpoints for adding and removing volumes on both VM and VMI objects.  Take a look at the functional test for an idea of how the AddVolume and RemoveVolume apis work in practice.

### CODE REVIEW

File: tests/vm_test.go [hunk 1]
```diff
@@ -243,6 +243,117 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() error {
+				updatedVM, err := virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &v12.GetOptions{})
+				if err != nil {
+					return err
+				}
+
+				if len(updatedVM.Status.VolumeRequests) > 0 {
+					return fmt.Errorf("waiting on all VolumeRequets to be processed")
```
Comment (awels) [inline tests/vm_test.go]: Typo in test, VolumeRequets

**Review by awels — COMMENTED** [review]
Looks great, found one little typo.

File: tests/vm_test.go (hunk 1 above)
Comment (davidvossel) [reply tests/vm_test.go]: fixed

File: pkg/virt-api/rest/subresource.go [hunk 2]
```diff
@@ -901,3 +901,301 @@ func (app *SubresourceAPIApp) FilesystemList(request *restful.Request, response
...
+	foundRemoveVol := false
+	for _, volume := range vmi.Spec.Volumes {
+		if volumeRequest.AddVolumeOptions != nil && volume.Name == volumeRequest.AddVolumeOptions.Name {
+			return "", fmt.Errorf("Unable to add volume [%s] because it already exists", volume.Name)
+		} else if volumeRequest.RemoveVolumeOptions != nil && volume.Name == volumeRequest.RemoveVolumeOptions.Name {
+			foundRemoveVol = true
+		}
+	}
+
+	if volumeRequest.RemoveVolumeOptions != nil && !foundRemoveVol {
```
Comment (awels) [inline pkg/virt-api/rest/subresource.go:982]: Wouldn't trying to remove something that doesn't exist be a no-op?

**Review by awels — COMMENTED** [review]
looks good one small question

**Pull request merged**